src/uct/api/v2/uct_v2.h

```diff
@@ -240,6 +251,18 @@ typedef enum {
 } uct_md_mem_dereg_flags_t;
 
 
+typedef enum {
+    /**
+     * Indicates that remote access to the returned remote key should fail if
+     * @ref uct_md_mem_dereg_v2 is called with the flag
+     * @ref UCT_MD_MEM_DEREG_FLAG_INVALIDATE on the memory handle. If this flag
+     * is not set, using the remote key after invalidating the memory region has
+     * an undefined behavior.
+     */
+    UCT_MD_MKEY_PACK_FLAG_INVALIDATE = UCS_BIT(0)
+} uct_md_mkey_pack_flags_t;
+
```

Comment:
Also, I don't see UCT_MD_MKEY_PACK_FLAG_INVALIDATE used anywhere in the code so this is a bit hard to decipher.

Reply:
the implementation is in #8002 ; it will create an indirect key (similar concept to memory window) and destroy it to invalidate